core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -173,33 +172,29 @@ public QueryBuilder parseInnerQueryBuilder() throws IOException {
      */
     @Nullable
     public QueryBuilder parseInnerFilterToQueryBuilder() throws IOException {
-        final boolean originalIsFilter = isFilter;
+        final boolean originalIsFilter = this.shardContext.isFilter;
         try {
-            isFilter = true;
+            this.shardContext.isFilter = true;
             return parseInnerQueryBuilder();
         } finally {
-            isFilter = originalIsFilter;
+            this.shardContext.isFilter = originalIsFilter;
         }
     }
 
     QueryBuilder parseInnerFilterToQueryBuilder(String queryName) throws IOException, QueryParsingException {
```

Comment:
one step further: I think we could deprecate/norelease these two parseFilter methods and make sure that our refactored queries don't use them, cause they have been moved to toFilter in the corresponding builder.